A9/cimentech

thiagolotufo opened the pull request:
## This solution refers to which of the apps?

A/M# - owasp-top10-2017-apps/a9/cimentech

## What did you do to mitigate the vulnerability?

This is a contribution to secDevLabs.
For a better use of exercise a9, I made some changes to the Docker Compose file, so that it is possible to update Drupal through lines of code.

## Did you test your changes? What commands did you run?

The vulnerability has not been mitigated :satisfied:

gustavocovas reviewed (changes requested):
Very nice, @thiagolotufo! This should simplify the solution for the Cimentech App.

Can we remove files that are obsolete in this repo? My understanding is that we do not need all folders under https://github.com/globocom/secDevLabs/tree/master/owasp-top10-2017-apps/a9/cimentech/app/html

Also, I don't think this [Dockerfile](https://github.com/globocom/secDevLabs/blob/master/owasp-top10-2017-apps/a9/cimentech/deployments/drupal.Dockerfile) is necessary.

thiagolotufo commented:
Hi @gustavocovas,

I removed the unnecessary files for the execution of this lab.
Could you check if everything is correct?

gustavocovas commented on owasp-top10-2017-apps/a9/cimentech/deployments/docker-compose.yml:
```diff
@@ -2,7 +2,7 @@ version: '3.3'
 services:
   drupal:
-    image: drupal:7.57
+    image: drupal:7.79
```
But this would fix the security vulnerability, and we do not want this!

gustavocovas commented on owasp-top10-2017-apps/a9/cimentech/deployments/drupal.Dockerfile:
```diff
@@ -1 +1 @@
-FROM drupal:7.57
\ No newline at end of file
+FROM drupal:7.79
```
Is this Dockerfile really necessary?

gustavocovas commented on owasp-top10-2017-apps/a9/cimentech/deployments/docker-compose.yml:
```diff
@@ -12,7 +12,8 @@ services:
       - 80:80
     volumes:
-      - ../app/html:/var/www/html
+      - ../app/html/sites/:/var/www/html/sites
```
Cool! We have a custom `robots.txt` (https://github.com/globocom/secDevLabs/blob/master/owasp-top10-2017-apps/a9/cimentech/app/html/robots.txt)  that appears in the readme: 

![image](https://user-images.githubusercontent.com/11429002/116296274-65bd1b80-a770-11eb-8b76-70001cd6ba8d.png)

Can we find a way to keep that?

thiagolotufo commented:
Hi @gustavocovas,

I updated docker compose file to a vulnerable version, and deleted unnecessary dockerfile.
Now, localhost/robots.txt can be accessed by a user.

Could you check if everything is correct?

thiagolotufo reviewed (commented):
> Nice! The `robots.txt` file is now different:
> 
> ![Screen Shot 2021-04-28 at 12 16 46](https://user-images.githubusercontent.com/11429002/116428953-c8b7bc80-a81b-11eb-87ba-277f5e11baa5.png)
> 
> I think this can be confusing for the user. Can we have the old one back (we would have to copy the custom one to the container) or update the Attack Narrative screenshot?

@gustavocovas 

I updated the README image. 
Now, the robots.txt image is correct.
